odoo/config.py

```diff
@@ -0,0 +1,1478 @@
+Option('reportgz', parse=strtobool, default=False)
+Option('root_path', file=False, parse=checkdir(os.R_OK), default=str(fullpath(Path(__file__)).parent))
+
+Option('population_size', args=["--size"], parse=str, action='store', default='small', help="Populate database with auto-generated data")
+Option('populate_models', args=["--models"], parse=CommaList.parser(str), action='append', default=CommaList(), metavar="MODEL OR PATTERN", help="List of model (comma separated or repeated option) or pattern")
+
+Option('verbose', args=['-v', '--verbose'], action='count', parse=int, default=0, help="Increase verbosity")
+Option('cloc_paths', args=['-p', '--path'], action='append', parse=CommaList.parser(checkpath('r')), default=CommaList(), metavar="PATH", help="File or directory")
+
+Option('deploy_module_path', args=["path"], parse=checkdir('r'), file=False, metavar="DIR", help="Path of the module to deploy")
+Option('deploy_url', args=["url"], parse=str, default='http://localhost:8069', nargs='?', metavar="URL", help="Url of the server")
+Option('deploy_db', args=["--db"], parse=str, help='Database to use if server does not use db-filter.')
+Option('deploy_login', args=["--login"], parse=str, default='admin', help="Login (default=admin)")
+Option('deploy_pwd', args=["--password"], parse=str, default='admin', help="Password (default=admin). Setting the password via the cli is a security risk, you may instead use the configuration file.")
+Option('deploy_verify_ssl', args=["--verify-ssl"], parse=strtobool, action='store_true', default=False, help="Verify SSL certificate")
+Option('deploy_force', args=["--force"], parse=strtobool, action='store_true', default=False, help="Force init even if module is already installed. (will update `noupdate=\"1\"` records)")
+
+Option('scaffold_modname', args=["name"], file=False, required=True, parse=str, metavar="MODULE", help="Name of the module to create")
+Option('scaffold_destdir', args=["dest"], nargs='?', default='.', parse=checkdir('w'), metavar="DIR", help="Directory to create the module in (default: current dir)")
+Option('scaffold_template', args=["-t", "--template"], parse=scaffold_template, default='default', metavar="TMPL", help="Use a custom module template, can be a template name or the path to a module template.")
+
+Option('startsc_path', args=["--path"], file=False, parse=checkdir('r'), default=fullpath('.'), envvar='VIRTUAL_ENV', metavar="DIRPATH", help="Directory where your project's modules are stored (will autodetect from current dir)")
+
+########################################################################
+#                                                                      #
+#                                GROUPS                                #
+#                                                                      #
+########################################################################
+
+@dataclasses.dataclass(frozen=True)
+class Group:
```

Comment:
seems like a weird way to make groups, wouldn't it make more sense to have them as a parameter of the `Option` class or even better, a context manager:

```python
with Group('my_group'):
    Option(...)
    Option(...)
```

Reply:
However, I realize that this is not compatible with the above comment on making Option definitions more declarative